api_tests/dry/peers_using_ip.ts

```diff
@@ -1,149 +1,118 @@
-import { expect, request } from "chai";
-import { ethers } from "ethers";
-import { Actor } from "../lib/actor";
+// These are stateless tests -- they don't require any state of the cnd and they don't change it
+// They are mostly about checking invalid request responses
+// These test do not use the sdk so that we can test edge cases
+import { threeActorTest, twoActorTest } from "../lib_sdk/actor_test";
+import "chai/register-should";
 import "../lib/setup_chai";
+import { expect, request } from "chai";
+import { Actor } from "../lib_sdk/actors/actor";
 import { sleep } from "../lib/util";
-
-(async () => {
-    const alice = new Actor("alice");
-    const bob = new Actor("bob");
-    const charlie = new Actor("charlie");
-    const alicePeerId = await alice.peerId();
-    const bobMultiAddress = bob.cndNetworkListenAddress();
-    const charliePeerId = await charlie.peerId();
-    const charlieMultiAddress = charlie.cndNetworkListenAddress();
-
-    describe("SWAP request with address", () => {
-        it("[Alice] Should not yet see Bob's peer id in her list of peers", async () => {
-            const res = await request(alice.cndHttpApiUrl()).get("/peers");
-
-            expect(res.status).to.equal(200);
-            expect(res.body.peers).to.be.empty;
-        });
-
-        it("[Alice] Should be able to make a swap request via HTTP api using a random peer id and Bob's ip address", async () => {
-            const res = await request(alice.cndHttpApiUrl())
-                .post("/swaps/rfc003")
-                .send({
-                    alpha_ledger: {
-                        name: "bitcoin",
-                        network: "regtest",
-                    },
-                    beta_ledger: {
-                        name: "ethereum",
-                        chain_id: 17,
-                    },
-                    alpha_asset: {
-                        name: "bitcoin",
-                        quantity: "100000000",
-                    },
-                    beta_asset: {
-                        name: "ether",
-                        quantity: ethers.utils.parseEther("10").toString(),
-                    },
-                    beta_ledger_redeem_identity:
-                        "0x00a329c0648769a73afac7f9381e08fb43dbea72",
-                    alpha_expiry:
-                        new Date("2080-06-11T23:00:00Z").getTime() / 1000,
-                    beta_expiry:
-                        new Date("2080-06-11T13:00:00Z").getTime() / 1000,
+import { createDefaultSwapRequest } from "../lib_sdk/utils";
+
+async function assertNoPeersAvailable(actor: Actor, message: string) {
```

Comment:
FYI: Chai actually has a way of providing your own assertions :)

Reply:
The way we currently did this was to create `assert` functions on the `Actor`, which would also work nicely here.